code/game/machinery/machinery.dm

```diff
@@ -243,6 +239,61 @@ Class Procs:
 
 ////////////////////////////////////////////////////////////////////////////////////////////
 
+/obj/machinery/mech_melee_attack(obj/mecha/M)
+	if(M.damtype == BRUTE || M.damtype == BURN)
+		visible_message("<span class='danger'>[M.name] has hit [src].</span>")
+		take_damage(M.force*2, M.damtype) // multiplied by 2 so we can hit machines hard but not be overpowered against mobs.
+		return 1
+	return 0
+
+/obj/machinery/attacked_by(obj/item/I, mob/living/user)
+	..()
+	take_damage(I.force, I.damtype, 1)
+
+/obj/machinery/proc/take_damage(damage, damage_type = BRUTE, sound_effect = 1)
+	switch(damage_type)
+		if(BRUTE)
+			if(sound_effect)
+				if(damage)
+					playsound(loc, 'sound/weapons/smash.ogg', 50, 1)
+				else
+					playsound(loc, 'sound/weapons/tap.ogg', 50, 1)
+		if(BURN)
+			if(sound_effect)
+				playsound(src.loc, 'sound/items/Welder.ogg', 100, 1)
```

Comment:
Can you remove the hitsound i added to atmospherics/portable/canister/attackby then?